odoo/addons/base/models/assetsbundle.py

```diff
@@ -348,6 +350,29 @@ def js(self):
 
         return js_attachment[0]
 
+    def xml(self):
+        if attachments := self.get_attachments('xml.js'):
+            return attachments[0]
+        bundle_content = textwrap.dedent(f"""
+            {{
+                const loadXmlTemplates = () => {{
+                    odoo.define("{self.name}.bundle.xml", ["@web/core/templates"], function (require) {{
+                        "use strict";
+                        const {{ checkPrimaryTemplateParents, registerTemplate, registerTemplateExtension }} = require("@web/core/templates");
+                        /* {self.name} */
+                        {self.generate_xml_bundle()}
+                    }});
+                }};
+                if (globalThis.odoo?.js_assets_loaded?.includes("{self.name}")) {{
```

Comment:
I think odoo should always be set, as it's defined in the `web.layout` template
```suggestion
                if (globalThis.odoo.js_assets_loaded?.includes("{self.name}")) {{
```

Reply:
Yep, I adapted all the custom places to make sure the `odoo` object is always created first, together with the `loadedJsAssets` entry.